rstudio-wsl/draft.Rmd

```diff
@@ -0,0 +1,157 @@
+---
+title: "Using RStudio Server on Windows"
+date: "Last updated on `r Sys.Date()`"
+output:
+  bookdown::html_document2:
+    keep_md: true
+    number_sections: false
+    toc: true
+toc-title: "Table of Contents"
+knit: "bookdown::render_book"
+---
+
+```{r klippy, echo=FALSE, include=TRUE}
+klippy::klippy(lang = c("bash", "r", "markdown"))
+
+knitr::opts_chunk$set(eval=FALSE, tidy = "styler")
+```
+
+## Introduction {#intro}
+
+This article explains about how to set up RStudio Server on Windows platform.
+
+As RStudio Server has primarily been designed to work with Linux distributions, Windows users must require a virtualization technology, called ["Windows Subsystem for Linux" (WSL)](https://docs.microsoft.com/en-us/windows/wsl/){target="_blank"}, to run a Linux kernel inside their system.
+
+## Prerequisite {#prereq}
+
+We recommend using the enhanced WSL (version 2) for setting up RStudio Server, which requires **Windows 10 (Version 2004; Build 19041 or higher)**.
+
+1. You can check your Windows version by pressing the `Windows logo key + R`, type `winver`, and hit enter.
+1. In case your system is lower than the suggested version, please [update to the latest Windows version](ms-settings:windowsupdate){target="_blank"}.
+
+* **Note:** For builds lower than 19041, WSL is not supported at all.
+
+## WSL Setup {#wsl-setup}
+
+Once you have met the [prerequisite](#prereq), you need to enable WSL features on your system.
+
+* **Note:** Use "Copy code" button below to easily copy all the required commands to clipboard and paste it directly into your PowerShell by pressing `Ctrl + V` key. Note that you have to hit enter key to execute the pasted code chunk in the console.
+
+1. Open "Windows PowerShell" with admin privileges.
+1. Turn on the WSL feature using the following command in the PowerShell:
+
+    ```{bash}
+    dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart
+    ```
+
+1. Enable the "Virtual Machine Platform" feature by running the following command in the PowerShell:
+
+    ```{bash}
+    dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart
+    ```
+
+1. Restart your system.
+1. Open "Windows PowerShell" with admin privileges again.
+1. Set WSL 2 as your default version using the following command in the PowerShell:
+
+    ```{bash}
+    wsl --set-default-version 2
+    ```
+
+1. Download and install [the latest WSL2 Linux kernel](https://wslstorestorage.blob.core.windows.net/wslblob/wsl_update_x64.msi) if you see the following message:
+
+    > WSL 2 requires an update to its kernel component. For information please visit `https://aka.ms/wsl2kernel`
+
+## Linux Distribution Setup {#linux-setup}
+
+After the [WSL 2 feature](#wsl-setup) has been enabled, you have to download and install a Linux distribution from [Microsoft Store](https://aka.ms/wslstore){target="_blank"}.
+
```

Comment:
Let's simplify and only discuss Ubuntu 20.04 in this article.

Some of the distros such as Kali Linux are not officially supported by RStudio, and although they may work, we should not suggest that we have tested them all on WSL.

So remove everything but the link for installing Ubuntu 20.04. You can mention that there are other distros, and that RStudio Server does support some of them, but details in this article are specific to Ubuntu 20.04.

More advanced users who are familiar with Linux and care about which distro they use will presumably have the skills to figure out the differences.